src/cryptoadvance/specter/helpers.py

```diff
@@ -241,16 +241,32 @@ def generate_mnemonic(strength=256):
     return words
 
 
+def wallet_type_by_derivation(derivation):
+    simple_derivation = derivation.replace("'", "").replace("h", "")
+    if simple_derivation.startswith("m/49"):
+        return "sh-wpkh"
+    elif simple_derivation.startswith("m/84"):
+        return "wpkh"
+    elif simple_derivation.startswith("m/48") and simple_derivation.endswith("/1"):
+        return "sh-wsh"
+    elif simple_derivation.startswith("m/48") and simple_derivation.endswith("/2"):
+        return "wsh"
+    else:
+        raise Exception(f"Unrecognized derivation: {derivation}")
```

Comment:
This function assumes that any derivation that starts with `84h` or even non-hardened `84` is `wpkh`, and the same for all other script types. This is not true - bips define very specific length for the key derivations:
BIP84 must have at least 3 **hardened** indexes - `m / purpose' / coin_type' / account' `. The same applies for BIP49. Multisig `48h` derivations must have length of 4.
You can use [`embit.bip32.parse_path()`](https://github.com/diybitcoinhardware/embit/blob/master/src/embit/bip32.py#L259) function to convert the path to a list and then check the length, first index, second index (must be `0x80000000` or `0x80000001`), that third index is hardened ( `>=0x80000000` )

Reply:
Now updated based on your comment further below.